nodejs/eks/nodegroup.ts

```diff
@@ -847,7 +847,6 @@ export function createManagedNodeGroup(name: string, args: ManagedNodeGroupOptio
     const nodeGroup = new aws.eks.NodeGroup(name, {
         ...nodeGroupArgs,
         clusterName: args.clusterName || core.cluster.name,
-        nodeGroupName: args.nodeGroupName || name,
```

Comment:
Ah, I thought in your original issue you posted, you were proposing something different, which was to not provide `nodeGroupName` if there's no arg passed in, but to keep it if there was. At least this way, this is _less_ breaking for folks who are explicitly passing in the arg.

Reply:
The `nodeGroupName` will be taken if provided from the `...nodeGroupArgs` spread. Passing the `nodeGroupName` argument will still work as before

Reply:
Ah, of course, sorry, yes. Thinking about this more, I guess one way to help folks who do rely on defaulting to the cluster name would be to indicate that they can still do this via a transformation. Let me ask one other person to take a look to make sure I'm not off in any of my thinking here.